kura/org.eclipse.kura.rest.service.listing.provider/src/main/java/org/eclipse/kura/internal/rest/service/listing/provider/RestServiceListingProvider.java

```diff
@@ -0,0 +1,186 @@
+package org.eclipse.kura.internal.rest.service.listing.provider;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.eclipse.kura.KuraErrorCode;
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.cloudconnection.request.RequestHandler;
+import org.eclipse.kura.cloudconnection.request.RequestHandlerRegistry;
+import org.eclipse.kura.internal.rest.service.listing.provider.dto.InterfacesIdsDTO;
+import org.eclipse.kura.internal.rest.service.listing.provider.dto.SortedServiceListDTO;
+import org.eclipse.kura.request.handler.jaxrs.DefaultExceptionHandler;
+import org.eclipse.kura.request.handler.jaxrs.JaxRsRequestHandlerProxy;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.useradmin.Role;
+import org.osgi.service.useradmin.UserAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("serviceListing/v1")
+public class RestServiceListingProvider {
+
+    private static final Logger logger = LoggerFactory.getLogger(RestServiceListingProvider.class);
+    private static final String REQUEST_DEBUG_MESSAGE = "Received request from: '{}'";
+
+    private static final String APP_ID_MQTT = "SERLIST-V1";
+    private static final String REST_ROLE = "kura.permission.rest.serviceListing";
```

Comment:
We can remove the specific `kura.permission.rest.serviceListing` role and allow access to this rest API to any authenticated user. If this is done this constant is not required.